File: Algorithms/Warmup/PlusMinus.java
package Warmup;

import java.util.Scanner;

/**
 *	Algorithms -> Warmup -> Plus Minus
 *	Easy (Super Easy)
 */
public class PlusMinus {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int arr[] = new int[n];
        for(int arr_i=0; arr_i < n; arr_i++){
            arr[arr_i] = in.nextInt();
        }
        in.close();
        double[] r = caclulatePlusMinus(arr);
        System.out.println(r[0]);
        System.out.println(r[1]);
        System.out.println(r[2]);
	}
	
	public static double[] caclulatePlusMinus(int[] arr) {
		double posCount = 0.0;
        double negCount = 0.0;
        double zeroCount = 0.0;
        int n = arr.length;
		for(int arr_i=0; arr_i < n; arr_i++){
            if(arr[arr_i] == 0) zeroCount++;
            if(arr[arr_i] > 0) posCount++;
            if(arr[arr_i] < 0) negCount++;
        }
		double[] r = {posCount/n, negCount/n, zeroCount/n};
		return r;
	}

}

File: Algorithms/Warmup/PlusMinus.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

/**
 *	Algorithms -> Warmup -> Plus Minus
 *	Easy (Super Easy)
 */
int main(){
    int n;
    scanf("%d",&n);
    int arr[n];
    for(int arr_i = 0; arr_i < n; arr_i++){
       scanf("%d",&arr[arr_i]);
    }

    double positiveCount = 0;
    double negativeCount = 0;
    for(int i = 0; i < n; i++) {
    	if(arr[i] != 0) {
    		if(arr[i] > 0) {
    			positiveCount += 1;
    		} else {
    			negativeCount += 1;
    		}
    	}
    }

    double fractionOfPositive = positiveCount / n;
    double fractionOfNegative = negativeCount / n;
    double fractionOfZero = (n - positiveCount - negativeCount) / n;

    printf("%0.4f\n%0.4f\n%0.4f", fractionOfPositive, fractionOfNegative, fractionOfZero);

    return 0;
}
